basic/src/test/java/org/jboss/hal/testsuite/test/runtime/server/configurationchanges/ConfigurationChangesTestCase.java

```diff
@@ -0,0 +1,184 @@
+import org.jboss.arquillian.graphene.page.Page;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.hal.testsuite.category.Shared;
+import org.jboss.hal.testsuite.creaper.ManagementClientProvider;
+import org.jboss.hal.testsuite.creaper.ResourceVerifier;
+import org.jboss.hal.testsuite.creaper.command.BackupAndRestoreAttributes;
+import org.jboss.hal.testsuite.page.runtime.ConfigurationChangesPage;
+import org.jboss.hal.testsuite.util.Console;
+import org.jboss.hal.testsuite.util.configurationchanges.ConfigurationChange;
+import org.jboss.hal.testsuite.util.configurationchanges.ConfigurationChangesResource;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.WebDriver;
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * @author Jan Kasik <jkasik@redhat.com>
+ *         Created on 9/14/16.
+ */
+
+@RunWith(Arquillian.class)
+@Category(Shared.class)
+public class ConfigurationChangesTestCase {
+
+    @Drone
+    private WebDriver browser;
+
+    @Page
+    private ConfigurationChangesPage page;
+
+    private static OnlineManagementClient client;
+    private static Operations operations;
+    private static Address serviceAddress;
+    private static ConfigurationChangesResource changesResource;
+
+    @BeforeClass
+    public static void beforeClass() {
+        client = ManagementClientProvider.createOnlineManagementClient();
+        operations = new Operations(client);
+        serviceAddress = client.options().isDomain ?
+                Address.host(client.options().defaultHost)
+                        .and("core-service", "management")
+                        .and("service", "configuration-changes") :
+                Address.coreService("management")
+                        .and("service", "configuration-changes");
+        changesResource = new ConfigurationChangesResource(client);
+    }
+
+    @Before
+    public void before() throws IOException, OperationException {
+        //make sure the service is up and running
+        if (operations.invoke("list-changes", serviceAddress).isFailed()) {
+            operations.add(serviceAddress, Values.of("max-history", 100)).assertSuccess();
+        }
+        page.navigate();
+    }
```

Comment:
You should make sure you are testing always the same => shouldn't you make sure that the max-history will always be 100 => even when list-changes op didn't fail?